### PR TITLE
Initial RPM packaging

### PR DESCRIPTION
**Pull request by jwendell**
This adds the make targets `rpm/istio` and `rpm/proxy` for creating rpm's for
Istio components. Artifacts will be created in `$ISTIO_OUT/rpm`.

### CODE REVIEW

**Comment by duderino** [comment]
@jwendell thank you, this is a great contribution.

@utka can you give it a review?

**Comment by duderino** [comment]
@jwendell also, please fix the lint and shellcheck issues so I can merge this once @utka signs off (if I merge as is then all subsequent PRs will fail these checks)

File: tools/packaging/common/istio-start.sh [hunk 1]
```diff
@@ -55,7 +55,7 @@ if [ -n "${ISTIO_CP_AUTH:-}" ]; then
 fi
 
 if [ -z "${ISTIO_SVC_IP:-}" ]; then
-  ISTIO_SVC_IP=$(hostname --ip-address)
+  ISTIO_SVC_IP=$(hostname --all-ip-addresses | cut -d ' ' -f 1)
```
Comment (costinm) [inline tools/packaging/common/istio-start.sh]: Some comment or explanation ? This seems a bit risky, is this option working on ubuntu/etc ?

**Comment by MrTrustor** [comment]
Ok, so I built the RPMs using this PR and tried to connect a CentOS 7 VM to a GKE cluster (1.12) with the Istio addon enabled (Istio 1.0.3). I managed to make it work, but there are 2 issues that If found:

1. The istio-node-agent package is bugged. The script `/usr/local/bin/istio-node-agent-start.sh` calls `/usr/local/bin/node_agent` when in fact it should call `/usr/local/bin/node-agent`. I manually patched the script, and it was fine.
2. Both the istio-node-agent and istio-pilot-agent packages need the user "istio-proxy" to be created. As far as I can see, that's not done anywhere in those packages. I had to create it manually with a `useradd -r istio-proxy`.

File: tools/packaging/common/istio-start.sh (hunk 1 above)
Comment (jwendell) [reply tools/packaging/common/istio-start.sh]: `--ip-address` does not work well on a centos vm with more than one ip address. `--all-ip-addresses` has more predictable output on all systems I've tested (debian, ubuntu, centos).

**Comment by jwendell** [comment]
@MrTrustor I had put the creation of the istio-proxy user only on pilot-agent, now it's on both. Also I fixed the node agent startup.

**Comment by jwendell** [comment]
@utka Do we have any currenctly active tests for VMs with debian?

**Comment by MrTrustor** [comment]
Did a new test with your modifications @jwendell, LGTM, thanks!

**Comment by utka** [comment]
Is there reason to merge it in release 1.1 and not master?

**Comment by jwendell** [comment]
@utka no strong reason. When I was working on this, 1.1 was the appropriate branch.

If you folks want it to be on master, I can rebase it on top of master and resend.

**Comment by utka** [comment]
Yes, master sounds better to me. Thank you!

**Comment by incfly** [comment]
@jwendell do you want to change this against to master and move forward?

**Comment by jwendell** [comment]
@incfly Sure, I'm about to do that today or tomorrow.

**Comment by jwendell** [comment]
Couldn't change my base branch, so, I just opened up a new PR: https://github.com/istio/istio/pull/13088

**Comment by jwendell** [comment]
I believe we can close this one in favor of #13088.

**Pull request closed without merging**